wallet/walletrpc.c

```diff
@@ -531,6 +531,58 @@ static const struct json_command listfunds_command = {
 };
 AUTODATA(json_command, &listfunds_command);
 
+static struct command_result *json_getbalance(struct command *cmd,
+					     const char *buffer,
+					     const jsmntok_t *obj UNNEEDED,
+					     const jsmntok_t *params)
+{
+    struct json_stream *response;
+    struct peer *p;
+    struct utxo **utxos;
+    struct amount_sat on_chain = AMOUNT_SAT(0), on_channels = AMOUNT_SAT(0);
```

Comment:
I would declare `struct amount_sat on_chain` and `struct amount_sat on_channels` without initializing them and init them not sooner than when required (line 551 and 561). This can help detect unwanted variable access by valgrind when someone else changes the code in the future.